### PR TITLE
Add a parameter "autorestart" to restart bot automatically if excepti…

### PR DESCRIPTION
**Pull request by iThom**
…on (default: False)

# Autorestart parameter

## Description

Parameter autorestart allow to autorestart the bot if an exception occured.
Autorestart is disabled by default.

## Type of change

Please make your selection.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Force some exceptions in code to test it.

### CODE REVIEW

File: pycryptobot.py [hunk 1]
```diff
@@ -988,13 +988,16 @@ def main(init=True):
 
         print(repr(e))
 
-        # Wait 1 second and try to relaunch application
-        time.sleep(1)
-        print('Restarting application after exception...')
+        if app.autoRestart():
```
Comment (letsautom8) [inline pycryptobot.py:991]: We should not be doing this as scheduling is handled 1 level higher and is sitting in pycryptobot.py.
We should put a try catch on scheduler and try invoke executeJob again 30 sec later.

**Review by letsautom8 — CHANGES_REQUESTED** [review]

File: pycryptobot.py (hunk 1 above)
Comment (whittlem) [reply pycryptobot.py:991]: I didn't see a response to this. How do you propose it is done?

Comment (iThom) [reply pycryptobot.py:991]: Sorry I don't have time now to change it, I'll try later to make the try/except around the s.run()

Comment (whittlem) [reply pycryptobot.py:991]: @letsautom8, what did you mean by your suggestion. Can you advise the correct way to do it?

Comment (whittlem) [reply pycryptobot.py:991]: Making the auto restart option optional would really help. What is happening with this PR?

Comment (iThom) [reply pycryptobot.py]: I did change to try/except around the scheduler.
Sorry for the delay I was not available.

Comment (whittlem) [reply pycryptobot.py]: It's okay, thanks very much for sorting this out.

**Pull request merged**